manifests/plugin/remote_execution/ssh.pp

```diff
@@ -54,39 +54,41 @@
     listen_on => $listen_on,
   }
 
-  if $ssh_kerberos_auth {
-    $kerberos_pkg = "${foreman_proxy::params::ruby_package_prefix}net-ssh-krb"
-    package { $kerberos_pkg:
-      ensure => present,
+  if $mode =~ /ssh/ {
```

Comment:
Isn't this always true? Both `ssh` and `ssh-async` match `/ssh/`. The patch will be a lot smaller if you don't need this additional indent.

Reply:
Felt like it would be cleaner to break the API first, THEN add support for `pull` and `pull-mqtt` modes

Follow up commit would append something like:

```puppet
} elsif $mode =~ /pull/ {
  if $mode == 'pull-mqtt' {
    # install and configure mqtt broker
  }

  # common configuration for 'pull' and 'pull-mqtt' modes, if any, goes here
}
```

Maybe it makes more sense for me to hold the indent until the follow up commit, too

Reply:
Made the more minimal change here to improve the clarity. Thanks for the suggestion